### PR TITLE
MOTOR-912 Update the Release Notes

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: RELEASE.rst [hunk 1]
```diff
@@ -63,6 +63,16 @@ and a `source distribution <https://packaging.python.org/guides/distributing-pac
 
      $ python3 -m twine upload dist/*
 
-#. Trigger a build of the docs on https://readthedocs.org/.
+#. Make sure the new version appears on https://motor.readthedocs.io/. If the
+    new version does not show up automatically, trigger a rebuild of "latest":
```
Comment (ShaneHarvey) [inline RELEASE.rst:67]: The formatting in most of these lines looks off. For example this line needs to be indented only 3 spaces.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: RELEASE.rst (hunk 1 above)
Comment (blink1073) [reply RELEASE.rst:67]: Fixed

File: RELEASE.rst [hunk 2]
```diff
@@ -63,6 +63,19 @@ and a `source distribution <https://packaging.python.org/guides/distributing-pac
...
+   https://readthedocs.org/projects/motor/builds/
 
-#. Announce!
+#. Bump the version number to <next version>.dev0 in setup.py/__init__.py,
+   commit, push.
+
+#. Publish the release version in Jira.
+
+#. Announce the release on:
+    https://developer.mongodb.com/community/forums/c/community/release-notes/
```
Comment (ShaneHarvey) [inline RELEASE.rst:76]: This line is still over indented. Also this link doesn't work for me, let's use: https://www.mongodb.com/community/forums/c/announcements/driver-releases

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: RELEASE.rst (hunk 2 above)
Comment (blink1073) [reply RELEASE.rst]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**